concepts/box/about.md

```diff
@@ -1,3 +1,153 @@
 # About
 
-TODO
+To understand [`Box`][box] it is helpful to understand what the [stack and heap][stack and heap] are.
+
+The stack is an area of memory that works like a last-in-first-out (LIFO) buffer. It is very fast to push data onto the stack and it is fast to pop it off.
+The data can include variables or function addresses, but the data must be of a known fixed size at compilation time.
+
+If the data is of an unknown size at compilation time or has a size that may change, then it must be allocated on the heap. The allocation process must find
+space in the heap big enough to store the data, and then the address of where that data starts in memory gets pushed onto the stack. Retrieving data from the
+heap is slower because we need to go through the address on the stack and follow that to the location of tha data on the heap.
+
+`Box` is essentially a smart pointer which allocates data on the heap. The address is on the stack, pointing to the data on the heap. There are several
+[reasons][reasons] to use `Box` to put data on the heap:
+
+- When you have a type whose size can’t be known at compile time and you want to use a value of that type in a context that requires an exact size
+
+- When you have a large amount of data and you want to transfer ownership but ensure the data won’t be copied when you do so
```

Comment:
So, you're saying that you have 

```rust
type GigArray = [u8; 1024 * 1024 * 1024];
```

In that case, it's true that it's more efficient to write

```rust
fn foo(ga: Box<GigArray>)
```

instead of 

```rust
fn foo(ga: GigArray)
```

However, that's because of a peculiarity of arrays in Rust; [arrays of any size implement `Copy` if the element type does](https://doc.rust-lang.org/std/primitive.array.html). In most other cases I can think of, including the same thing with vectors, Rust doesn't necessarily copy the data when transferring ownership; that was a fairly obvious optimization for the compiler team, and it's been in the language for a while.

I'm not sure that this point is worth including; it's rare to transfer ownership of large arrays.